src/hazelcore/Builtins.re

```diff
@@ -0,0 +1,190 @@
+/*
+   Built-in functions for Hazel.
+
+   To add a built-in function or constant, write the implementation in the
+   `Impls` module below and add it to the `built-ins` list.
+
+   See the existing ones for reference.
+ */
+
+open Sexplib.Std;
+
+[@deriving sexp]
+type args = list(DHExp.t);
+
+[@deriving sexp]
+type eval = (args, DHExp.t => EvaluatorResult.t) => EvaluatorResult.t;
+
+[@deriving sexp]
+type elab = DHExp.t;
+
+[@deriving sexp]
+type t = {
```

Comment:
Naming is maybe still a bit off. This type should probably be called Builtin.t. Then you can define a Builtins.t module which actually defines the specific builtins that we are defining for Hazel right now.

Reply:
Alright, that sounds good. `mod` and `PI` I added for making sure currying and constants work. Do we want to keep those? And what other built-ins might be wanted for now?